lib/govuk_app_config/govuk_statsd.rb

```diff
@@ -8,7 +8,7 @@ module GovukStatsd
 
   def self.client
     @client ||= begin
-      statsd_client = ::Statsd.new("localhost")
+      statsd_client = ::Statsd.new(ENV["STATSD_HOST"] || "localhost", 8125, ENV["STATSD_PROTOCOL"]&.to_sym || :udp)
```

Comment:
It looks like we've already got a precedent for a prefix of `GOVUK_` to statsd env vars (see line below) so given they're set for this gem rather than the underlying statsd gem.

Should the Statsd class in the fork we've created (statsd-ruby-tcp) have a different name to `::Statsd` - It strikes me that if any other apps that use this gem also have a dependency on Statsd we'll have a namespace collision - It's a shame we can't point to a git branch at a dependency of a gem like we'd do in an app.

It does seem a little odd that we don't have an env var for PORT. I guess it's a point of need thing, but it ends up reading strange on this utility.

I had a somewhat whacky idea that could maybe use an env var of `GOVUK_STATSD_URI` that could be set and then parse that for args

For example:

```
statsd_uri = URI.parse(ENV.fetch("GOVUK_STATSD_URI", "localhost"))
statsd_client = ::Statsd.new(statsd_uri.host, stats.port || 8125, statsd_uri.scheme&.to_sym || :udp)
```

Reply:
Thanks! I updated the env var names.

I am open to enabling apps to configure the port, but I don't think we need this yet given our apps aren't using Statsd's admin port `8126` - if they do need to, we could make this configurable. Would that be OK?

> we'll have a namespace collision

Thanks for mentioning that. Search API uses both `govuk_app_config` and `statsd-ruby`.

We could stop wrapping the Statsd client with GovukStatsd, but that's a bit nuclear. I've changed the name in the temporary gem to `StatsdTcp`, since I'm hoping I'll get my PR merged in soon.

Reply:
Nice one. No love for my single env var idea? 😢 

Thanks for re-releasing the Gem with the new class name 👍

Reply:
> No love for my single env var idea? 😢

It's a neat idea! I can definitely imagine a scenario where an app would want to send messages to the admin port (8126). Setting `GOVUK_STATSD_URI=localhost:8126` via puppet might be neat.

But setting `GOVUK_STATSD_URI=localhost:8125` might be problematic if an app also wants to send messages to the stats port (8125). We might have to add `GOVUK_STATSD_ADMIN_URI=localhost:8126`, duplicating the host.

Another reason is that I don't foresee the port numbers we'd use for Statsd changing on a per-app basis, since we'd also have to make that change at Statsd.